Exadel.Compreface/Clients/FaceDetectionClient.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Exadel.Compreface.Clients.Config;
+using Exadel.Compreface.Configuration;
+using Exadel.Compreface.Helpers;
+using Exadel.Compreface.Services;
+using Flurl.Http;
+using Microsoft.Extensions.Configuration;
+using System.Text.Json;
+
+namespace Exadel.Compreface.Clients
+{
+    public class FaceDetectionClient
+    {
+        public FaceDetectionService FaceDetectionService { get; private set; }
+
+        public FaceDetectionClient(string apiKey, string domain, string port) : this(new ComprefaceConfiguration(apiKey, domain, port))
+        { }
+
+        public FaceDetectionClient(IConfiguration configuration, string sectionForApiKey, string sectionForDomain, string sectionForPort) : this(new ComprefaceConfiguration(configuration, sectionForApiKey, sectionForDomain, sectionForPort))
+        { }
+
+        public FaceDetectionClient(ComprefaceConfiguration configuration)
+        {
+            var apiClient = new ApiClient();
+
+            ConfigInitializer.InitializeApiKeyInRequestHeader(configuration.ApiKey);
+            ConfigInitializer.InitializeSnakeCaseJsonConfigs();
```

Comment:
it is better to have some kind of convention, so either create new objects in the first line then second lines will be for this function calls or otherwise flip the order. Example:

```
var apiClient = new ApiClient();

FaceDetectionService = new FaceDetectionService(apiClient: apiClient, configuration: configuration);

ConfigInitializer.InitializeApiKeyInRequestHeader(configuration.ApiKey);
ConfigInitializer.InitializeSnakeCaseJsonConfigs();
```

Reply:
The queue of calls is like it was in your ComprefaceClient class. Could you please explain what is this convention? Because according to the Clean Code book we choose how to arrange groups of code by some logic. And in both variants we have logic that separate, but just we have different order))